src/matrix/sync_room_message.rs

```diff
@@ -25,6 +25,41 @@ use crate::matrix::verification::handle_verification_request;
 /// https://url.spec.whatwg.org/#fragment-percent-encode-set
 const FRAGMENT: &AsciiSet = &CONTROLS.add(b' ').add(b'"').add(b'<').add(b'>').add(b'`');
 
+async fn generate_fresh_file(dir: PathBuf, filename: &str) -> Result<(tokio::fs::File, String)> {
+    let fresh_filepath = tokio::task::spawn_blocking({
+        let filename = String::from(filename);
+        move || -> Result<PathBuf> {
+            let firstdotidx = filename.find('.');
+            let (prefix, suffix) = match firstdotidx {
+                Some(idx) => filename.split_at(idx),
+                None => (filename.as_str(), ""),
+            };
+            let (_, tmpfile) = tempfile::Builder::new()
+                .prefix(&(prefix.to_owned() + "-"))
+                .suffix(&suffix)
+                .tempfile_in(dir)?
+                .keep()?;
```

Comment:
`.keep()` gives you a file which can be converted into a `tokio::fs::File` with from, without re-opening:
```
use anyhow::Result;
use tokio::io::AsyncWriteExt;

#[tokio::main]
async fn main() -> Result<()> {
    let (f, p) = tempfile::Builder::new()
        .prefix("foo-")
        .suffix(".txt")
        .tempfile_in(".")?
        .keep()?;
    let mut tf = tokio::fs::File::from(f);
    tf.write_all(b"test").await?;

    println!("Hello: {:?}", p);
    Ok(())
}
```
So I guess you could use it like that if you want something a bit shorter?

Reply:
Looks good, done. Thanks for the suggestion!